### PR TITLE
bug #201: Handle no user agent header on 403

### PR DESCRIPTION
**Pull request by NizamLZ**

### CODE REVIEW

**Comment by NizamLZ** [comment]
Circle CI says this :( :

image

Due to dependency cache maybe?

**Comment by phatuous** [comment]
Interesting. I tried pulling the this PR, and it runs fine locally on my Mac. might be some sort of issue with node module dependencies being cached?

**Comment by issa-tseng** [comment]
yeah, it is indeed a dependency cache issue. i'll have it rerun without the cache for you.

File: lib/http/endpoint.js [hunk 1]
```diff
@@ -222,7 +222,7 @@ const defaultEndpoint = endpointBase({
 });
 
 
-////////////////////////////////////////
+///////////////////////////////////////
```
[the remark this answers is not in the document]
Comment (NizamLZ) [reply lib/http/endpoint.js]: Pardon me. I did that in a desperate attempt to trigger the pipeline for the 2nd time.

File: lib/http/endpoint.js [hunk 2]
```diff
@@ -179,7 +179,7 @@ const defaultResultWriter = (result, request, response) => {
 // the necessary work to translate that error into an HTTP error and send it out.
 const defaultErrorWriter = (error, request, response) => {
   if (error.isProblem === true) {
-    if ((error.httpCode === 403) && (request.get('user-agent').match(/tableau/i))) {
+    if ((error.httpCode === 403) && request.get('user-agent') && (request.get('user-agent').match(/tableau/i))) {
```
[the remark this answers is not in the document]
Comment (NizamLZ) [reply lib/http/endpoint.js]: I have updated it as requested.

**Comment by issa-tseng** [comment]
perfect, thanks!

**Comment by issa-tseng** [comment]
might i interest you in #211 and/or #216? :)

**Comment by NizamLZ** [comment]
@issa-tseng Sure. Will look into it as soon as I get some time.

**Pull request merged**